controllers/idler/idler_controller_test.go

```diff
@@ -682,51 +682,89 @@ func TestAppNameTypeForControllers(t *testing.T) {
 	}
 }
 
-func TestAppNameTypeForInidividualPods(t *testing.T) {
+func TestNotificationAppNameTypeForPods(t *testing.T) {
 	//given
 	idler := &toolchainv1alpha1.Idler{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "alex-stage",
+			Name: "feny-stage",
 			Labels: map[string]string{
-				toolchainv1alpha1.SpaceLabelKey: "alex",
+				toolchainv1alpha1.SpaceLabelKey: "feny",
 			},
 		},
 		Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
 	}
+	namespaces := []string{"dev", "stage"}
+	usernames := []string{"feny"}
+	nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "feny", "advanced", "abcde11", namespaces, usernames)
+	mur := newMUR("feny")
 
-	t.Run("Test AppName/Type in notification", func(t *testing.T) {
-		namespaces := []string{"dev", "stage"}
-		usernames := []string{"alex"}
-		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
-		mur := newMUR("alex")
-		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet, mur)
-		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
-		p := preparePayloadsSinglePod(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo).standalonePods[0]
-		// first reconcile to track pods
-		res, err := reconciler.Reconcile(context.TODO(), req)
-		assert.NoError(t, err)
-		assert.True(t, res.Requeue)
+	testpod := map[string]struct {
+		pcond           corev1.PodCondition
+		expectedAppType string
+	}{
+		"Individual-Completed-Pod": {
+			pcond:           corev1.PodCondition{Type: "Ready", Reason: "PodCompleted"},
+			expectedAppType: "Pod",
+		},
```

Comment:
Instead of relying on the test name below:
```
if pt == "Individual-Completed-Pod" || pt == "Individual-NonCompleted-Pod" {
```
I would add a dedicated field to distinguish controlled owned pods from standalone ones. Something like this: 
```suggestion
		"Individual-Completed-Pod": {
			pcond:           corev1.PodCondition{Type: "Ready", Reason: "PodCompleted"},
			expectedAppType: "Pod",
			expectedNotificationCreated: true, // set to true if it's expected that a notification is created
			controllerOwned: false,
		},
```
And then use it:
```
if tcs.controllerOwned {
	preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond)
} else {
	preparePayloadsSinglePod(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond)
}
```
and
```
if tcs.expectedNotificationCreated {
	...
} else {
	...
}

Reply:
done